core/src/main/java/com/linecorp/armeria/server/RejectedPathMappingHandler.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A handler that is invoked when a {@link ServerBuilder} rejects to bind a {@link Service} at
+ * a certain {@link PathMapping}. For example, the following code will trigger this handler:
+ *
+ * <pre>{@code
+ * ServerBuilder sb = new ServerBuilder();
+ * sb.service("/hello", serviceA);
+ * sb.service("/hello", serviceB); // Tried to bind at the same path mapping again.
+ * sb.build();
+ * }</pre>
+ *
+ * @see ServerBuilder#rejectedPathMappingHandler(RejectedPathMappingHandler)
+ */
+@FunctionalInterface
+public interface RejectedPathMappingHandler {
+
+    /**
+     * A {@link RejectedPathMappingHandler} that logs a warning message for a problematic {@link PathMapping}.
+     */
+    RejectedPathMappingHandler WARN = (virtualHost, mapping, existingMapping) -> {
+        final Logger logger = LoggerFactory.getLogger(RejectedPathMappingHandler.class);
+        final String a = mapping.toString();
+        final String b = existingMapping.toString();
+        final String hostnamePattern = virtualHost.hostnamePattern();
+        if (a.equals(b)) {
+            logger.warn("Virtual host '{}' has a duplicate path mapping: {}",
```

Comment:
Since this only really happens once, maybe we don't need to worry about the efficiency and could just share the same code that uses concatenation for constructing the error message.

Reply:
We are still in Java 8 and we can't have a private method in an interface, so let me leave this as it is for now rather than introducing an extra utility class. Will add a todo. :-)